src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -1357,19 +1357,35 @@ void EmitExpressionConditional(RegexNode node)
 
                 // The first child node is the conditional expression.  If this matches, then we branch to the "yes" branch.
                 // If it doesn't match, then we branch to the optional "no" branch if it exists, or simply skip the "yes"
-                // branch, otherwise. The conditional is treated as a positive lookahead if it isn't already one.
+                // branch, otherwise. The conditional is treated as a positive lookahead.
                 RegexNode conditional = node.Child(0);
-                if (conditional is { Type: RegexNode.Require })
+                if (conditional.Type == RegexNode.Require)
```

Comment:
any particular reason to write this way rather than the "is" way?

Reply:
Not a big difference.  There's an unnecessary null check in the first and I'm just used to writing it as the second. This is actually a squash of multiple commits, one where I deleted the optimization and then one where I wrote it back,  and I reverted to my normal style when doing so.

Reply:
I mean like `if (conditional is not { Type: RegexNode.Require })`

Ah, the is/is not adds a null check.

Reply:
oops, comments crossed.